tardis/tests/tests_slow/test_w7.py

```diff
@@ -48,90 +51,128 @@ def setup(self, reference, data_path, atomic_data_fname):
-                self.obtained_radial1d_model.last_line_interaction_angstrom)
+                self.result.last_line_interaction_angstrom)
 
     def test_nubar_estimators(self):
         assert_allclose(
                 self.reference['nubar_estimators'],
-                self.obtained_radial1d_model.nubar_estimators)
+                self.result.nubar_estimators)
 
     def test_ws(self):
         assert_allclose(
                 self.reference['ws'],
-                self.obtained_radial1d_model.ws)
+                self.result.ws)
 
     def test_luminosity_inner(self):
         assert_quantity_allclose(
                 self.reference['luminosity_inner'],
-                self.obtained_radial1d_model.luminosity_inner)
+                self.result.luminosity_inner)
 
     def test_spectrum(self):
-        assert_quantity_allclose(
+        try:
+            assert_quantity_allclose(
                 self.reference['luminosity_density_nu'],
-                self.obtained_radial1d_model.spectrum.luminosity_density_nu)
+                self.result.spectrum.luminosity_density_nu)
 
-        assert_quantity_allclose(
+            assert_quantity_allclose(
                 self.reference['delta_frequency'],
-                self.obtained_radial1d_model.spectrum.delta_frequency)
+                self.result.spectrum.delta_frequency)
 
-        assert_quantity_allclose(
+            assert_quantity_allclose(
                 self.reference['wavelength'],
-                self.obtained_radial1d_model.spectrum.wavelength)
+                self.result.spectrum.wavelength)
 
-        assert_quantity_allclose(
+            assert_quantity_allclose(
                 self.reference['luminosity_density_lambda'],
-                self.obtained_radial1d_model.spectrum.luminosity_density_lambda)
+                self.result.spectrum.luminosity_density_lambda)
+
+            self.plot_spectrum(passed=True)
+        except Exception as e:
+            self.plot_spectrum(passed=False)
+            raise e
+
+    def plot_spectrum(self, passed):
+        plt.suptitle("Deviation in spectrum_quantities", fontweight="bold")
+
+        # `ldl_` prefixed variables associated with `luminosity_density_lambda`.
+        # Axes of subplot are extracted, if we wish to make multiple plots
+        # for different spectrum quantities all in one figure.
+        ldl_ax = plt.subplot(111)
+        ldl_ax.set_title("Deviation in luminosity_density_lambda")
+        ldl_ax.set_xlabel("Wavelength")
+        ldl_ax.set_ylabel("Relative error (1 - obtained / baseline)")
+        deviation = 1 - (
+            self.result.spectrum.luminosity_density_lambda.value /
+            self.reference['luminosity_density_lambda'].value)
+
+        if passed:
+            ldl_ax.text(0.8, 0.8, 'passed', transform=ldl_ax.transAxes,
+                        bbox={'facecolor': 'green', 'alpha': 0.5, 'pad': 10})
+            ldl_ax.plot(self.reference['wavelength'], deviation, "g+")
```

Comment:
what's the difference between 'g+' and 'rx'? can you please use a keyword argument so it's clear what option this argument refers to?